server/src/account/account.service.ts

```diff
@@ -384,10 +387,30 @@ export class AccountService {
     }
   }
 
-  async findOneGiftCode(code: string, used = false): Promise<GiftCode | null> {
+  async findOneGiftCode(
```

Comment:
remove condition, return result directly, and judge by fields `used` & `expiredAt`